### PR TITLE
feat(geography): outline basic driver model and types

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/geography/src/drivers/interfaces/geography-service.interface.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	/**
+	 * Retrieves the list of countries available to the given store.
+	 */
+	list(): DaffCountry[];
+
+	/**
+	 * Retrieve precise information about a specific country.
+	 */
+	get(id: DaffCountry['id']): DaffCountry;
+}
```
Comment (griest024) [inline libs/geography/src/drivers/interfaces/geography-service.interface.ts]: We need an injection token as well.

File: libs/geography/src/models/subdivision.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * The subdivisions of a country as per ISO-3166-2
+ * For the United States this is commonly termed as "States"
+ */
+export interface DaffSubdivision {
+  id: string;
+  name: string;
+  name_en: string;
```
Comment (griest024) [inline libs/geography/src/models/subdivision.ts]: Do you think that `name_en` field should be nullable? Magento for example does not provide a separate english name field for [Region](https://devdocs.magento.com/guides/v2.3/graphql/queries/directory-countries.html#region). LMK what you think.

File: libs/geography/src/drivers/interfaces/geography-service.interface.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/geography/src/drivers/interfaces/geography-service.interface.ts]: fixed

File: libs/geography/src/models/subdivision.ts (hunk 2 above)
Comment (damienwebdev) [reply libs/geography/src/models/subdivision.ts]: fixed

**Pull request merged**